Assets/MixedRealityToolkit.Services/InputSimulation/Implementation/MouseRotationProvider.cs

```diff
@@ -0,0 +1,147 @@
+    {
+        private bool isRotating = false;
+        public bool IsRotating => isRotating;
+
+        private static int numRotating = 0;
+        private static bool isMouseJumping = false;
+        private static bool wasCursorVisible = true;
+
+        public void Update(KeyBinding rotationKey, KeyBinding cancelRotationKey, bool toggle)
+        {
+            bool wasRotating = isRotating;
+
+            // Only allow the mouse to control rotation when Unity has focus.
+            // This enables the player to temporarily alt-tab away without having the player
+            // look around randomly back in the Unity Game window.
+            if (!Application.isFocused)
+            {
+                isRotating = false;
+            }
+            else
+            {
+                if (toggle)
+                {
+                    if (isRotating)
+                    {
+                        // pressing escape will stop capture
+                        isRotating = !KeyInputSystem.GetKeyDown(cancelRotationKey);
+                    }
+                    else
+                    {
+                        // any kind of click will capture focus
+                        isRotating = KeyInputSystem.GetKeyDown(rotationKey);
+                    }
+                }
+                else
+                {
+                    isRotating = KeyInputSystem.GetKey(rotationKey);
+                }
+            }
+
+            if (!wasRotating && isRotating)
+            {
+                OnStartRotating(rotationKey);
+            }
+            else if (wasRotating && !isRotating)
+            {
+                OnEndRotating(rotationKey);
+            }
+        }
+
+        private static void OnStartRotating(KeyBinding rotationKey)
+        {
+            if (numRotating == 0)
+            {
+                if (rotationKey.BindingType == KeyBinding.KeyType.Mouse)
+                {
+                    // if mousebutton is either left, right or middle
+                    SetWantsMouseJumping(true);
+                }
+                else if (rotationKey.BindingType == KeyBinding.KeyType.Key)
+                {
+                    // if mousebutton is either control, shift or focused
+                    UnityEngine.Cursor.lockState = CursorLockMode.Locked;
+                    // save current cursor visibility before hiding it
+                    wasCursorVisible = UnityEngine.Cursor.visible;
+                    UnityEngine.Cursor.visible = false;
+                }
+
+                // do nothing if (this.MouseLookButton == MouseButton.None)
+            }
+
+            ++numRotating;
+        }
+
+        private static void OnEndRotating(KeyBinding rotationKey)
+        {
+            --numRotating;
+
+            if (numRotating == 0)
+            {
+                if (rotationKey.BindingType == KeyBinding.KeyType.Mouse)
+                {
+                    // if mousebutton is either left, right or middle
```

Comment:
>left, right or middle [](start = 48, length = 21)

Was this a comment from an older iteration? The helper function doesn't seem to have anything specific to left/right/middle mouse button

Reply:
Same comment on the thing down below:

// if mousebutton is either control, shift or focused

---
In reply to: [320515680](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5853#discussion_r320515680) [](ancestors = 320515680)

Reply:
Legacy comment, fixed.